src/Tribe/Extension_Loader.php

```diff
@@ -0,0 +1,242 @@
+<?php
+/**
+ * Class Tribe__Extension_Loader
+ */
+class Tribe__Extension_Loader {
+
+	/**
+	 * Plugin header data
+	 *
+	 * @var array {
+	 *      Plugin header data
+	 *
+	 *      @param array $plugin_basename Plugin header key/value pairs.
+	 * }
+	 */
+	private $plugin_data = array();
+
+	/**
+	 * Class instance.
+	 *
+	 * @var Tribe__Extension_Loader The singleton instance.
+	 */
+	private static $instance;
+
+	/**
+	 * Returns the singleton instance of this class.
+	 *
+	 * @return Tribe__Extension_Loader instance.
+	 */
+	public static function instance() {
+		return null === self::$instance ? new self() : self::$instance;
+	}
+
+	/**
+	 * Intializes each extension.
+	 */
+	private function __construct() {
+		$prefixes = self::get_extension_file_prefixes();
+		$extension_filepaths = self::get_plugins_with_prefix( $prefixes );
+
+		foreach ( $extension_filepaths as $plugin_file ) {
+			$this->instantiate_extension( $plugin_file );
+		}
+	}
+
+	/**
+	 * Get list of active plugins with a given prefix in the plugin folder path.
+	 *
+	 * @param string|array $prefix Prefixes you want to retrieve.
+	 *
+	 * @return array List of plugins with prefix in path.
+	 */
+	public static function get_plugins_with_prefix( $prefix ) {
+		$plugin_list = wp_get_active_and_valid_plugins();
+
+		if ( is_multisite() ) {
+			$plugin_list = array_merge( $plugin_list, wp_get_active_network_plugins() );
+		}
+
+		$extension_list = array();
+
+		foreach ( $plugin_list as $plugin ) {
+			$base = plugin_basename( $plugin );
+
+			if ( self::strpos_array( $base, $prefix ) === 0 ) {
+				$extension_list[] = $plugin;
+			}
+		}
+
+		return $extension_list;
+	}
+
+	/**
+	 * Gets the plugin data from the plugin file header
+	 *
+	 * @param string $plugin_file Absolute path to plugin file containing header.
+	 *
+	 * @see get_plugin_data() for WP Admin only function this is similar to.
+	 *
+	 * @return array Plugin data; keys match capitalized file header declarations.
+	 */
+	public static function get_plugin_data( $plugin_file ) {
+		$default_headers = array(
+			'Name' => 'Plugin Name',
+			'PluginURI' => 'Plugin URI',
+			'Version' => 'Version',
+			'ExtensionClass' => 'Extension Class',
+			'ExtensionFile' => 'Extension File',
+			'Description' => 'Description',
+			'Author' => 'Author',
+			'AuthorURI' => 'Author URI',
+			'TextDomain' => 'Text Domain',
+			'DomainPath' => 'Domain Path',
+			'Network' => 'Network',
+		);
+
+		return get_file_data( $plugin_file, $default_headers, 'plugin' );
+	}
+
+	/**
+	 * Behaves exactly like strpos(), but accepts an array of needles.
+	 *
+	 * @see strpos()
+	 *
+	 * @param string       $haystack String to search in.
+	 * @param array|string $needles  Strings to search for.
+	 * @param int          $offset   Starting position of search.
+	 *
+	 * @return false|int Integer position of first needle occurrence.
+	 */
+	public static function strpos_array( $haystack, $needles, $offset = 0 ) {
```

Comment:
That's a good idea. On second review it gave me some ideas. I wonder if I can break a few other things out:
- Tribe__Extension_Loader::get_plugin_data() could probably be made into tribe_get_plugin_data(), and used in Tribe__Extension::get_plugin_data() as well.
- Tribe__Extension::search_var() was handed to me by Gustavo, something very similar is used in a class he recently wrote. I wonder if that function, and a setting function based on Tribe__Extension::set(), would make sense somewhere? Perhaps in a a Tribe__Array_Utils class? And we could event put the strpos_array() in there if we wanted. What do you think?
- I think Tribe__Extension::filter_meta_links() and Tribe__Extension::add_meta_link() probably make sense inside of a Tribe__Plugin_Meta_Links class. 

I'd be happy to alter things a bit to make this happen. It might take an hour or two for something we technically have no budget for. But, if you approve I can squeeze this in before midday tomorrow so it can go through code review.